tools/RepositoryManagement/DevHome.RepositoryManagement/ViewModels/RepositoryManagementItemViewModel.cs

```diff
@@ -2,61 +2,451 @@
+            {
+                Directory.Delete(myDirectory);
+            }
+
+            File.SetAttributes(ClonePath, FileAttributes.Normal);
+            Directory.Delete(ClonePath, false);
+        }
+
+        var repository = GetRepositoryReportIfNull(nameof(DeleteRepository));
+        if (repository == null)
+        {
+            // Do not warn the user here.  If the repository is not in the database
+            // the repository management page will not display the repository
+            // when entities are fetched.
+            return;
+        }
+
+        _dataAccess.RemoveRepository(repository);
     }
 
     [RelayCommand]
-    public void MakeConfigurationFileWithThisRepository()
+    public async Task MakeConfigurationFileWithThisRepository()
     {
-        throw new NotImplementedException();
+        try
+        {
+            // Show the save file dialog
+            using var fileDialog = new WindowSaveFileDialog();
+
+            // TODO: Needs Localization
+            fileDialog.AddFileType(_stringResource.GetLocalized("{0} file", "YAML"), ".winget");
+            fileDialog.AddFileType(_stringResource.GetLocalized("{0} file", "YAML"), ".dsc.yaml");
+            var fileName = fileDialog.Show(_window);
+
+            // If the user selected a file, write the configuration to it
+            if (!string.IsNullOrEmpty(fileName))
+            {
+                var repositoryToUse = _dataAccess.GetRepository(RepositoryName, ClonePath);
+                var repository = GetRepositoryReportIfNull(nameof(MakeConfigurationFileWithThisRepository));
+                if (repository == null)
+                {
+                    return;
+                }
+
+                var configFile = _configurationFileBuilder.MakeConfigurationFileForRepoAndGit(repositoryToUse);
+                await File.WriteAllTextAsync(fileName, configFile);
+            }
+        }
+        catch (Exception e)
+        {
+            _log.Error(e, $"Failed to download configuration file.");
+        }
     }
 
     [RelayCommand]
-    public void OpenFileExplorerToConfigurationsFolder()
+    public void RunConfigurationFile()
+    {
+        var repository = GetRepositoryReportIfNull(nameof(RunConfigurationFile));
+        if (repository == null)
+        {
+            return;
+        }
+
+        if (!repository.HasAConfigurationFile)
+        {
+            _log.Warning($"The repository with name {RepositoryName} and clone location {ClonePath} does not have a configuration file.");
+            return;
+        }
+
+        var configurationFileLocation = repository.ConfigurationFileLocation ?? string.Empty;
+        var processStartInfo = new ProcessStartInfo();
+        processStartInfo.UseShellExecute = true;
+        processStartInfo.FileName = "winget";
+        processStartInfo.ArgumentList.Add("configure");
+        processStartInfo.Verb = "RunAs";
+
+        StartProcess(processStartInfo, nameof(RunConfigurationFile));
```

Comment:
Should we be using the DSC service Amir added?

Reply:
We cannot.  The DSC service does not know ahead of time why it needs to run as admin.  Instead we run CMD as admin.  This is done in [CloneRepoSummaryInformationFileViewModel](https://github.com/microsoft/devhome/blob/main/tools/SetupFlow/DevHome.SetupFlow/ViewModels/CloneRepoSummaryInformationFileViewModel.cs#L110)